src/util/psbt/serialize.rs

```diff
@@ -89,6 +89,30 @@ impl Deserialize for PublicKey {
     }
 }
 
+impl Serialize for EcdsaSig {
+    fn serialize(&self) -> Vec<u8> {
+        let mut buf = Vec::with_capacity(72);
+        buf.extend(self.sig.serialize_der().iter());
```

Comment:
Note: `extend_from_slice` not needed because `extend` is already specialized. Unimportant: could've been `&` instead of `.iter()` I think.

Reply:
Not working that way: `serialize_der` returns `SerializedSignature`, which is not an iterator - and requires `&*` prefix to deref first into slice and than get iterator over it. IMO having this prefix is much uglier than calling `iter()`.

Reply:
Ah, thought it's `[u8; N]`. Would be nice to `impl IntoIterator for SerailizedSignature`